src/ADAL.PCL.Desktop/WebUI.cs

```diff
@@ -47,9 +47,17 @@ internal abstract class WebUI : IWebUI
             var sendAuthorizeRequest = new Action(
                 delegate
                 {
+                    string requestId = Telemetry.GetInstance().CreateRequestId();
+                    Telemetry.GetInstance().StartEvent(requestId, "ui_event");
+
+                    UIEvent UiEvent = new UIEvent();
+
                     authorizationResult = this.Authenticate(authorizationUri, redirectUri);
+
+                    Telemetry.GetInstance().StopEvent(requestId, UiEvent, "ui_event");
                 });
```

Comment:
Yes this is in accordance with the discussion we had. I have made the start and stop event as a part of the delegate sendAuthorizeRequest. So whenever this delegate is called in the code(eg:- line 69) it will be able to measure the ui_time.This is more efficient as we have UI_Event hook at only one place in the code rather than having it in multiple places.